pkg/authnz/authconfig/oidc.go

```diff
@@ -79,75 +84,108 @@
 			http.Error(w, "no session", http.StatusUnauthorized)
 			return
 		}
-		url := oauthConfig.AuthCodeURL(oauthStateString)
-		http.Redirect(w, r, url, http.StatusTemporaryRedirect)
+		// 2. Client sends the request to the Authorization Server.
+		authCodeURL := oauthConfig.AuthCodeURL(oauthStateString)
+		http.Redirect(w, r, authCodeURL, http.StatusTemporaryRedirect)
 	}
 }
 
-func (c *OIDCConfig) callbackHandler(runtime *authruntime.ProviderRuntime, oauthConfig *oauth2.Config, provider *oidc.Provider) http.HandlerFunc {
+func (c *OIDCConfig) callbackHandler(runtime *authruntime.ProviderRuntime, oauthConfig *oauth2.Config,
+	provider *oidc.Provider, verifier *oidc.IDTokenVerifier) http.HandlerFunc {
+
 	return func(w http.ResponseWriter, r *http.Request) {
+		// 3. Authorization Server Authenticates the End-User.
+		// 4. Authorization Server obtains End-User Consent/Authorization.
+		// 5. Authorization Server sends the End-User back to the Client with an Authorization Code.
+
 		s, err := runtime.GetSession(r)
 		if err != nil {
 			http.Error(w, "no session", http.StatusBadRequest)
 			return
 		}
 
+		// Make sure the returned state matches the one saved in the session cookie to prevent CSRF attacks
 		state := r.FormValue("state")
-		if s.Nonce == nil || *s.Nonce != state {
-			http.Error(w, "bad nonce", http.StatusBadRequest)
+		if s.Nonce == nil {
+			http.Error(w, fmt.Sprintf("bad state value: expected = nil, got = %s", state), http.StatusBadRequest)
```

Comment:
He, I thought that something like this would be possible. Nice to see that the code scanning stuff works.
The change will be dropped again anyway, I don't think the expected value should be printed either, and it's not worth going through the trouble of escaping the state from the query parameter.